airbyte-integrations/connectors/source-file/source_file/client.py

```diff
@@ -363,3 +367,126 @@ def streams(self) -> Iterable:
             "properties": self._stream_properties(),
         }
         yield AirbyteStream(name=self.stream_name, json_schema=json_schema, supported_sync_modes=[SyncMode.full_refresh])
+
+
+class ArchiveClient:
+    """Class that manages reading and parsing data from streams"""
+
+    reader_class = URLFile
+
+    def __init__(self, url: str, provider: dict, archive_format: str, file_format: str = 'csv', reader_options: str = '{}', file_rename: str = '{}'):
+        self._url = url
+        self._provider = provider
+        self._untar_destination = 'local'
+        self._archive_format = archive_format
+
+        try:
+            self._file_format = json.loads(file_format)
+        except json.decoder.JSONDecodeError as err:
+            self._file_format = {'all': (file_format or 'csv')}
+        else:
+            self._file_format = self._file_format or {'all' : 'csv'}
+        finally:
+            assert len(self._file_format.values())
+        
+        try:
+            self._reader_options = json.loads(reader_options)
+        except json.decoder.JSONDecodeError as err:
+            error_msg = f"Failed to parse reader options {repr(err)}\n{reader_options}\n{traceback.format_exc()}"
+            logger.error(error_msg)
+            raise ConfigurationError(error_msg) from err
+        else:
+            self._reader_options = self._reader_options or {'all' : {}}
+
+        try:
+            self._file_rename = json.loads(file_rename)
+        except json.decoder.JSONDecodeError as err:
+            self._file_rename = {'all' : file_rename}
+        else:
+            self._file_rename = self._file_rename or {'all' : 'tablename'}
+        
+
+        if self._archive_format == 'Not Archived':
+            self._file_format = {'all' : next(iter(self._file_format.values()))}
+            self._file_rename = {'all' : next(iter(self._file_rename.values()))}
+            self._reader_options = {'all' : next(iter(self._reader_options.values()))}
+
+    @property
+    def reader(self) -> reader_class:
+        return self.reader_class(url=self._url, provider=self._provider)
+
+    @property
+    def binary_source(self):
+        binary_formats = {"excel", "feather", "parquet", "orc", "pickle"}
+        return (
+            (self._archive_format != 'Not Archived') or 
+            (self._file_format['all'] in binary_formats)
+        )
```

Comment:
These properties are already declared in Client class. You could leverage inheritance.